test/Microsoft.Net.Http.Headers.Tests/EntityTagHeaderValueTest.cs

```diff
@@ -100,6 +100,52 @@ public void Equals_UseSameAndDifferentETags_EqualOrNotEqualNoExceptions()
         }
 
         [Fact]
+        public void CompareStrongly_UseSameAndDifferentETags_EqualOrNotEqualNoExceptions()
+        {
+            var etag1 = new EntityTagHeaderValue("\"tag\"");
+            var etag2 = new EntityTagHeaderValue("\"TAG\"");
+            var etag3 = new EntityTagHeaderValue("\"tag\"", true);
+            var etag4 = new EntityTagHeaderValue("\"tag1\"");
+            var etag5 = new EntityTagHeaderValue("\"tag\"");
+            var etag6 = EntityTagHeaderValue.Any;
+
+            Assert.False(EntityTagHeaderValue.CompareStrongly(etag1, etag2), "Different casing.");
+            Assert.False(EntityTagHeaderValue.CompareStrongly(etag2, etag1), "Different casing.");
+            Assert.False(EntityTagHeaderValue.CompareStrongly(etag1, null), "tag vs. <null>.");
+            Assert.False(EntityTagHeaderValue.CompareStrongly(null, etag1), "<null> vs tag.");
+            Assert.False(EntityTagHeaderValue.CompareStrongly(null, null), "<null> vs. <null>.");
+            Assert.False(EntityTagHeaderValue.CompareStrongly(etag3, etag3), "weak vs. weak.");
+            Assert.False(EntityTagHeaderValue.CompareStrongly(etag1, etag3), "strong vs. weak.");
+            Assert.False(EntityTagHeaderValue.CompareStrongly(etag3, etag1), "weak vs. strong.");
+            Assert.False(EntityTagHeaderValue.CompareStrongly(etag1, etag4), "tag vs. tag1.");
+            Assert.False(EntityTagHeaderValue.CompareStrongly(etag1, etag6), "tag vs. *.");
+            Assert.True(EntityTagHeaderValue.CompareStrongly(etag1, etag5), "tag vs. tag.");
+        }
```

Comment:
Make it theory so that all the cases are tested no matter how many of them are failed. In the current form, if the first assertion filed the following assertions won't be executed.